libraries/Microsoft.Bot.Connector/Authentication/PasswordServiceClientCredentialFactory.cs

```diff
@@ -102,6 +102,9 @@ public override Task<ServiceClientCredentials> CreateCredentialsAsync(string app
             }
         }
 
+        /// <summary>
+        /// TODO this is internal implementation does not need to reply on the public class.
```

Comment:
> TODO  [](start = 11, length = 6)

We should not leave this here.

Reply:
Yes, it took me a minute to even remember what this meant. What I was attempting to say here when I added this comment a couple of months back is that in this refactoring there is actually no need to have a public surface area like MicrosoftAppCredentials.

I'd actually go further and suggest that if you are having to make your concrete classes public then you have failed to abstract the instantiation in a factory. ServiceClientCredentialsFactory is such a factory, so actually all you need is public factory methods rather than public class hierarchies. In fact our old code was full of conditional logic testing an constructing the right concrete type - essentially nullifying whatever benefits we might have gained from factoring teh logic into classes in the first place.